### PR TITLE
Scala 3 build test

### PR DESCRIPTION
**Pull request by eloots**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes #xxxx

## Purpose

Adapt build test CI for Scala 3. Change to have one line per module

## Background Context

Make CI script easier to read and update

## References

Prepare for Scala 3 support #11260

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -170,8 +170,10 @@ jobs:
       #  Play / Test / compile
       #  "
       cmd: >-
-        sbt -Dscala.version=$MATRIX_SCALA scalaVersion
-        "all Sbt-Routes-Compiler/compile Play-Streams/compile"
+        sbt -Dscala.version=$MATRIX_SCALA scalaVersion "
+          Sbt-Routes-Compiler/Test/compile;
+          Play-Streams/Test/compile;
```
Comment (octonato) [inline .github/workflows/build-test.yml:175]: If I'm not mistaking, we saw that the tests in the routes compiler was not compiling, no?

Comment (eloots) [reply .github/workflows/build-test.yml]: I ran the tests locally before opening this PR and they passed. Which is also confirmed by the CI run...

Comment (octonato) [reply .github/workflows/build-test.yml]: Perfect! 
I don't know I thought it was not the case. Maybe we have warnings that I wrongly interpreted as errors.

Comment (eloots) [reply .github/workflows/build-test.yml]: CI passes for Scala 3 specific stuff but fails in scripted tests. What do we do in such case @octonato ? Merge anyway ?

Comment (octonato) [reply .github/workflows/build-test.yml]: Is failing consistently or is that a flaky test? 

If failing consistently, we can't merge

**Pull request merged**